packages/tsconfck/src/parse.js

```diff
@@ -48,9 +49,12 @@ export async function parse(filename, options) {
 		let result;
 		if (filename !== tsconfigFile && cache?.hasParseResult(tsconfigFile)) {
 			result = await getParsedDeep(tsconfigFile, cache, options);
+			if (result.referenced?.then) {
```

Comment:
this is needed otherwise `pnpm bench` fails.  However it shouldn't be. getParsedDeep awaits referenced, so i am not sure why it can still be a promise here...

Reply:
I can't understand why this is neither either. I think you could add a TODO to review this later? It won't harm perf for now